File: LeetCode75/Backspace_String_Compare/Backspace_String_Compare.cc
// 844. Backspace String Compare C++
#include <string>
#include <vector>

class Solution {
public:
    bool backspaceCompare(std::string s, std::string t) {
        std::vector<char> newS;
        std::vector<char> newT;
        int lenS = s.size();
        int lenT = t.size();
        for (int i = 0; i < lenS; i++) {
            if (s[i] == '#') {
                if (!newS.empty()) newS.pop_back();
            } else newS.emplace_back(s[i]);
        }
        for (int i = 0; i < lenT; i++) {
            if (t[i] == '#') {
                if (!newT.empty()) newT.pop_back();
            } else newT.emplace_back(t[i]);
        }
        return newS == newT;
    }
};

File: LeetCode75_LevelOne/Backspace_String_Compare/Backspace_String_Compare.py
# 844. Backspace String Compare Python

class Solution(object):
    def backspaceCompare(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        newS = []
        newT = []
        for cs in s: newS = newS + [cs] if cs != '#' else newS[:-1]
        for ct in t: newT = newT + [ct] if ct != '#' else newT[:-1]
        return newS == newT
